src/main/java/com/synopsys/integration/phonehome/request/PhoneHomeRequestBodyBuilder.java

```diff
@@ -39,6 +40,30 @@
     private final List<String> artifactModules = new ArrayList<>();
     private final Map<String, String> metaData = new HashMap<>();
 
+    public static PhoneHomeRequestBodyBuilder createForBlackDuck(String integrationRepoName, String registrationId, String blackDuckServerUrl, String integrationVersion, String blackDuckVersion) {

```

Comment:
Since we're ditching the suppliers, I'd love some other way to express to consumers of the library that they should pass in either the version if they get it, or null / StringUtils.EMPTY if they don't. We want solutions to avoid trying to come up with their own "unknown version"

Reply:
@Nullable wins the day.